src/haddock/modules/scoring/sasascore/__init__.py

```diff
@@ -0,0 +1,153 @@
+from haddock.libs.libparallel import get_index_list
+from haddock.modules.scoring.sasascore.sasascore import (
+    AccScore,
+    AccScoreJob,
+    prettify_df,
+    )
+from haddock.modules.analysis import (
+    get_analysis_exec_mode,
+    )
+from typing import Optional
+
+RECIPE_PATH = Path(__file__).resolve().parent
+DEFAULT_CONFIG = Path(RECIPE_PATH, "defaults.yaml")
+
+
+class HaddockModule(BaseHaddockModule):
+    """HADDOCK3 module to perform accessibility scoring."""
+
+    name = RECIPE_PATH.name
+
+    def __init__(self,
+                 order: int,
+                 path: Path,
+                 initial_params: FilePath = DEFAULT_CONFIG) -> None:
+        super().__init__(order, path, initial_params)
+
+    @classmethod
+    def confirm_installation(cls) -> None:
+        """Confirm module is installed."""
+        return
+
+    def _rearrange_output(self, output_name: FilePath, path: FilePath,
+                          ncores: int) -> None:
+        """Combine different tsv outputs in a single file."""
+        output_fname = Path(path, output_name)
+        self.log(f"rearranging output files into {output_fname}")
+        key = output_fname.stem.split(".")[0]
+        # Combine files
+        with open(output_fname, 'w') as out_file:
+            for core in range(ncores):
+                tmp_file = Path(path, f"{key}_" + str(core) + ".tsv")
+                with open(tmp_file) as infile:
+                    out_file.write(infile.read())
+                tmp_file.unlink()
+        self.log(f"Completed reconstruction of {key} files.")
+        self.log(f"{output_fname} created.")
+    
+    def _run(self) -> None:
+        """Execute module."""
+        try:
+            models_to_score = self.previous_io.retrieve_models(
+                individualize=True
+                )
+        except Exception as e:
+            self.finish_with_error(e)
+
+        nmodels = len(models_to_score)
+        # index_list for the jobs with linear scaling
+        ncores = parse_ncores(n=self.params['ncores'], njobs=nmodels)
+        idx_list = get_index_list(nmodels, ncores)
+
+        # loading buried and accessible residue dictionaries
+        buried_resdic = {
+            key[-1]: value for key, value
+            in self.params.items()
+            if key.startswith("resdic_buried")
+            }
+        acc_resdic = {
+            key[-1]: value for key, value
+            in self.params.items()
+            if key.startswith("resdic_accessible")
+            }
+        # remove _
+        buried_resdic.pop("_")
+        acc_resdic.pop("_")
+        # finding the chains
+        buried_violations_chains = [f"acc_{ch}" for ch in acc_resdic.keys()]
+        acc_violations_chains = [f"bur_{ch}" for ch in buried_resdic.keys()]
+        violations_chains = acc_violations_chains + buried_violations_chains
+        # initialize jobs
+        sasascore_jobs: list[AccScoreJob] = []
+
+        for core in range(ncores):
```

Comment:
I'm a bit confused here, why are you doing this per core and why not per model?

Reply:
we don't want to launch hundreds/thousands of extremely short jobs, but rather concatenate them so that each core is dealing with a decent number of jobs

Reply:
This is already handled by the scheduler:

https://github.com/haddocking/haddock3/blob/01eb9ae3e3c97b06248a9a90ca414d2f41c70d5f/src/haddock/libs/libparallel.py#L148

Reply:
@rvhonorato I have been thinking about this. I don't think this way of handling the tasks is a good idea (related also to the caprieval code linked above). If the running mode is not local and less_io is not active (the if clause in the caprieval output), this still amounts at creating a huge amount of files (for example 10k in the higher sampling scenarios), which can be avoided by simply parallelising internally on the number of cores (much lower in general). Additionally, for super-short tasks like this one, this probably makes things faster. what do you think?

Reply:
I just found out that pattern is found in many other places in the code, see here: https://github.com/haddocking/haddock3/issues/970

So please add this one to the list there and I can handle them in another PR

Reply:
We don't need to think about `less_io` unless it's a CNS-based module, what you are doing here is duplicating the work of the scheduler - even if you split the tasks here, when they arrive at the scheduler they are re-split, so no need to do it twice and better to let the lower function do it